website/integrations: add Actual budget

nicedevil007 opened the pull request:
## Details

I created a new documentation for Actual Budget. This got a recent update. Now it supports OpenID Connect :)

---

## Checklist

-   [X] Local tests pass (`ak test authentik/`)
-   [X] The code has been formatted (`make lint-fix`)

If applicable

-   [X] The documentation has been updated
-   [X] The documentation has been formatted (`make website`)

nicedevil007 committed:
Update sidebarsIntegrations.js

Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
Create index.mdx for Actual Budget

Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

dominic-r reviewed (changes requested):
Hey, thanks for your contribution. Just a few pointers to get this MR going. I'll do a more thorough review once some of these issues are fixed as I will have a better understanding / overview of this guide.

Also, I noticed the referenced UI elements are code-blocked. They should be put in bold as per the [authentik developer documentation](https://docs.goauthentik.io/docs/developer-docs/docs/style-guide#fonts-and-font-styling).

Please do let me know if you have any questions.

nicedevil007 commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,77 @@
+- `actual.company` is the FQDN of the Actual Budget install.
+- `authentik.company` is the FQDN of the authentik install.
+
+## authentik configuration
```
Heya, good morning 😊 thank you for reviewing my PR again!

Is there an example or template on how the documentation pages should look like after we all should use the wizard instead of the manual steps?

Just oriented myself now the last PR for Semaphore UI 😂

nicedevil007 replied:
Is this mandatory for this PR? I would do it for my next guide, I don't want to start from scratch again 😭

Learned something new how git works and what I should try to avoid next time (node modules 😂 f.e.)

I guess this guide is close to the same structure as my other guides I already have written here an you reviewed, at least I hope so 😊

tanberry replied:
Let's leave it as it is for now, and change it to use the Wizard later. The Wizard is still in Preview, anyway... so when we make it the default then we can update all of the Integration guides. :-)

Thanks @nicedevil007 for this guide! Let me review quickly (thanks @4d62 for your edits!) and let's get this merged...

nicedevil007 commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,77 @@
+- If you are redirected back to the `https://actual.company` URL and can see the budget file selection page, the setup was successful.
+
+:::info
+Users are not automatically created when logging in with authentik. The owner must manually create each user. To do so, click `Server online` at the top next to your name and select `User Directory`. Add a new user. The `Username` must match the one in authentik. You can now grant the new user access to your budget by clicking `Server online` next to your name at the top and selecting `User Acces ...
```
That is correct. First one is the king.

nicedevil007 replied:
That is correct. First one is the king.

nicedevil007 commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,77 @@
+- Name: `SP-actual`
+- Client type: `Confidential`
+- Redirect URIs/Origins (RegEx): `https://actual.company/openid/callback`
+- Signing Key: `authentik Self-signed Certificate`
```
In discord of actual was a recent post about OpenID authentication and most guys got problems with right signing keys. The solution for them was to switch back to authentik's self signed cert.

nicedevil007 commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,77 @@
+
+## Actual Budget configuration
+
+1. Log into your Actual Budget using a browser of your choice and open your budget.
```
The guy got a list of all budgets and it just simple click on the name/entry of the budget. It is designed like a button with the name of the budget inside.

nicedevil007 committed:
Update website/integrations/services/actual-budget/index.mdx

Co-authored-by: 4d62 <git@sdko.org>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
Changes made for comments of @4d62

Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
warning about first user

Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
make lint & make website...

nicedevil007 committed:
removed old dev stuff changes

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+
+> Actual Budget is a super-fast and privacy-focused app for managing your finances. At its core is the well-proven and much-loved envelope budgeting methodology.
+> You own your data and can do whatever you want with it. It features multi-device sync, optional end-to-end encryption, and much more.
+>
```
Normally I would rewrite the markety-hype description of the product, and state plainly what the product does, common use cases, etc. However, since this is a direct quote from the product's website, we can leave it for now. As a future project, we will probably rewrite all of them.

nicedevil007 replied:
I can do it for this PR as well, we don't have to wait 😂

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+Actual Budget supports the RS256 algorithm. Please ensure to consider this when choosing the appropriate signing key.
+:::
+
+Take note of the Client ID and Client Secret; you will need to provide them to Actual Budget in Step 3.
```
Yay, a semi-colon used properly! :-)

nicedevil007 replied:
ChatGPT helped me a bit 😎 as might already know, English is not my native language 😂

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+6. Enter the variables provided by authentik here.
+    - Set `OpenID Provider` to `Authentik`
+    - Set `OpenID provider URL` to `https://authentik.company/application/o/actual/`
+    - Set `Client ID` to `<client-id>` from Step 2
```
```suggestion
    - Set `Client ID` to `<client-id>`
```

tanberry replied:
Do you mean Step 2.. under the authentik configuration section (which is not numbered with steps) or ...?

nicedevil007 replied:
Yes but your suggestion is absolutely more understandable 💪

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+    - Set `OpenID Provider` to `Authentik`
+    - Set `OpenID provider URL` to `https://authentik.company/application/o/actual/`
+    - Set `Client ID` to `<client-id>` from Step 2
+    - Set `Client secret` to `<client-secret>` from Step 2
```
```suggestion
    - Set `Client secret` to `<client-secret>` to the value shown in the authentik provider.
```

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+    - Set `Client secret` to `<client-secret>` from Step 2
+
+:::warning
+The first user to log into Actual Budget via OpenID will become the owner and administrator with the highest privileges for the budget. Check the info box at the bottom for more information on how to create additional users.
```
```suggestion
The first user to log into Actual Budget via OpenID will become the owner and administrator with the highest privileges for the budget. For more information on how to create additional users, see the Note below.
```

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+## Test the login
+
+- Open a browser of your choice and navigate to `https://actual.company`.
+- Select the OpenID login method in the dropdown menu and click the **Sign in with OpenID** button.
```
```suggestion
- Select the OpenID login method in the dropdown menu and click **Sign in with OpenID**.
```

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+- If you are redirected back to the `https://actual.company` URL and can see the budget file selection page, the setup was successful.
+
+:::info
+Users are not automatically created when logging in with authentik. The owner must manually create each user. To do so, click `Server online` at the top next to your name and select `User Directory`. Add a new user. The `Username` must match the one in authentik. You can now grant the new user access to your budget by clicking `Server online` next to your name at the top and selecting `User Acces ...
```
```suggestion
Users are not automatically created when logging in with authentik. The owner must manually create each user in Actual Budget. To do so, click `Server online` at the top next to your name and select `User Directory`. Add a new user. The `Username` must match the one in authentik. You can now grant the new user access to your budget by clicking `Server online` next to your name at the top and selecting `User Access`.
```

tanberry reviewed (changes requested):
OK, a few edits, but overall looks great. We really appreciate your adding another integration guide, @nicedevil007 !

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
Remove step 3 and marketing Blabla

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+
+1. Sign in to Actual Budget with a browser of your choice and access your budget by clicking on its name.
+
+2. Click your budget in the top-left corner to open the dropdown menu and select `Settings`.
```
```suggestion
2. Click your budget in the top-left corner to open the dropdown menu and select **Settings**.
```

tanberry replied:
Yes, now I see what 4d62 was saying.... the UI components, the buttons and field names, etc., should be bolded, not in code format.

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+
+2. Click your budget in the top-left corner to open the dropdown menu and select `Settings`.
+
+3. Scroll to the bottom and select `Show advanced settings`. Scroll again and select `I understand the risks, show experimental features`.
```
```suggestion
3. Scroll to the bottom and select **Show advanced settings**. Scroll again and select **I understand the risks, show experimental features**.
```

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+
+3. Scroll to the bottom and select `Show advanced settings`. Scroll again and select `I understand the risks, show experimental features`.
+
+4. To enable the option `OpenID authentication method` select the checkbox next to it.
```
```suggestion
4. To enable the option **OpenID authentication method** select the checkbox next to it.
```

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+
+4. To enable the option `OpenID authentication method` select the checkbox next to it.
+
+5. Scroll up to the new option `Authentication method...` and click the `Start using OpenID` button.
```
```suggestion
5. Scroll up to the new option **Authentication method...** and click **Start using OpenID**.
```

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+5. Scroll up to the new option `Authentication method...` and click the `Start using OpenID` button.
+
+6. Enter the variables provided by authentik here.
+    - Set `OpenID Provider` to `Authentik`
```
```suggestion
    - Set **OpenID Provider** to `Authentik`
```

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+
+6. Enter the variables provided by authentik here.
+    - Set `OpenID Provider` to `Authentik`
+    - Set `OpenID provider URL` to `https://authentik.company/application/o/actual/`
```
```suggestion
    - Set **OpenID provider URL** to `https://authentik.company/application/o/actual/`
```

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+6. Enter the variables provided by authentik here.
+    - Set `OpenID Provider` to `Authentik`
+    - Set `OpenID provider URL` to `https://authentik.company/application/o/actual/`
+    - Set `Client ID` to `<client-id>` to the value shown in the authentik provider.
```
```suggestion
    - Set **Client ID** to `<client-id>` (the value shown in the authentik provider)
```

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+    - Set `OpenID Provider` to `Authentik`
+    - Set `OpenID provider URL` to `https://authentik.company/application/o/actual/`
+    - Set `Client ID` to `<client-id>` to the value shown in the authentik provider.
+    - Set `Client secret` to `<client-secret>` to the value shown in the authentik provider.
```
```suggestion
    - Set **Client secret** to `<client-secret>` (the value shown in the authentik provider)
```

tanberry commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+- If you are redirected back to the `https://actual.company` URL and can see the budget file selection page, the setup was successful.
+
+:::info
+Users are not automatically created when logging in with authentik. The owner must manually create each user in Actual Budget. To do so, click `Server online` at the top next to your name and select `User Directory`. Add a new user. The `Username` must match the one in authentik. You can now grant the new user access to your budget by clicking `Server online` next to your name at the top and sele ...
```
```suggestion
Users are not automatically created when logging in with authentik. The owner must manually create each user in Actual Budget. To do so, click **Server online** at the top next to your name and select **User Directory**. Add a new user. The `Username` must match the one in authentik. You can now grant the new user access to your budget by clicking **Server online** next to your name at the top and selecting **User Access**.
```

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 commented:
Ahh I got it, changed all to bold now. I have one question now as I read through the docs again.

Shouldn't this part:

6. Enter the variables provided by authentik here.
    - Set **OpenID Provider** to `Authentik`
    - Set **OpenID provider URL** to `https://authentik.company/application/o/actual/`
    - Set **Client ID** to `<client-id>` (the value shown in the authentik provider)
    - Set **Client secret** to `<client-secret>` (the value shown in the authentik provider)

then look like this?

6. Enter the variables provided by authentik here.
    - Set **OpenID Provider** to **Authentik**
    - Set **OpenID provider URL** to https://_authentik.company_/application/o/_actual_/
    - Set **Client ID** to _client-id_ (the value shown in the authentik provider)
    - Set **Client secret** to _client-secret_ (the value shown in the authentik provider)

According to this: 
![image](https://github.com/user-attachments/assets/7a0a5aa4-41d9-409c-b62e-27f567f37346)

And yes I guess then I have to rewrite a bit more.

But if I remember right, most documentations here are done with the code stuff "``"

nicedevil007 commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+5. Scroll up to the new option **Authentication method...** and click **Start using OpenID**.
+
+6. Enter the variables provided by authentik here.
+    - Set **OpenID Provider** to `Authentik`
```
Unfortunately it is writte with a capital A on their page and it is hardcoded :/ right now it is `Authentik`

![image](https://github.com/user-attachments/assets/c5bf3309-83fb-4a56-981c-d2164f0619ae)

nicedevil007 replied:
should I change it to `authentik` and raise an issue there :D

nicedevil007 commented on website/integrations/services/actual-budget/index.mdx:
```diff
@@ -0,0 +1,82 @@
+
+5. Scroll up to the new option **Authentication method...** and click **Start using OpenID**.
+
+6. Enter the variables provided by authentik here.
```
the last part was added because you adviced me to add something that this part is not overread. So what should I do now. Maybe I missunderstood your comment from here? https://github.com/goauthentik/authentik/pull/12590#discussion_r1906181053

dominic-r reviewed (approved):
Not-that-important thing I noticed earlier. looks good overall. Thanks for your contribution.

nicedevil007 committed:
Update website/integrations/services/actual-budget/index.mdx

Co-authored-by: 4d62 <git@sdko.org>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

tanberry commented:
> Ahh I got it, changed all to bold now. I have one question now as I read through the docs again.
> 
> Shouldn't this part:
> 
> 6. Enter the variables provided by authentik here.
>    
>    * Set **OpenID Provider** to `Authentik`
>    * Set **OpenID provider URL** to `https://authentik.company/application/o/actual/`
>    * Set **Client ID** to `<client-id>` (the value shown in the authentik provider)
>    * Set **Client secret** to `<client-secret>` (the value shown in the authentik provider)
> 
> then look like this?
> 
> 6. Enter the variables provided by authentik here.
>    
>    * Set **OpenID Provider** to **Authentik**
>    * Set **OpenID provider URL** to https://_authentik.company_/application/o/_actual_/
>    * Set **Client ID** to _client-id_ (the value shown in the authentik provider)
>    * Set **Client secret** to _client-secret_ (the value shown in the authentik provider)
> 
> According to this: ![image](https://private-user-images.githubusercontent.com/17103076/401408266-7a0a5aa4-41d9-409c-b62e-27f567f37346.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY0NjIyNzgsIm5iZiI6MTczNjQ2MTk3OCwicGF0aCI6Ii8xNzEwMzA3Ni80MDE0MDgyNjYtN2EwYTVhYTQtNDFkOS00MDljLWI2MmUtMjdmNTY3ZjM3MzQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA5VDIyMzI1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM5ZmFhNGMzZGFmZWE3MzRmM2RlNzQzNzMyZjJhMDEzM2FjMzM4Zjk5ODY0ZjAzNWY1MDI5M2NjZjIyYjQ5NTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tweCKPLRtcZ5XM_q1usGiRsTUmZqBPD8ziTygWsjW4s)
> 
> And yes I guess then I have to rewrite a bit more.
> 
> Oh, and finally, a big thank you to both of you for helping me improve my understanding of Git with every pull request! I truly appreciate it!
> 
> But if I remember right, most documentations here are done with the code stuff "``"

Ui elements are bolded, values (such as the URL) are in code text, and variables are italicized. So yes, your rewriet is good, except for not sure why Authentik is bolded. But perhas in the Actual Budget UI it is a selection... it's fine how it is!

nicedevil007 committed:
hope that helps a bit => step 6

remove the part about client id and secret with description that it is from authentik because step 6 clears this already up and during the setup of the provider in authentik there is a hint that ID and secret is need later in this guide

Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

tanberry reviewed (approved):
Thanks again for this contribution, @nicedevil007 ! It looks really good, and thanks for your patience with my and @4d62 's constant edits, lol. 

I know what you mean about the more you use git the more clear it becomes, so yeah this PR has been good practice I imagine. I am curious what you mentioned about modules, node modules? What was that exactly? Are you combining multiple repos using modules?

I think we can merge this now!

nicedevil007 committed:
Update index.mdx

now with italic instead of codestuff for placeholders or variables

Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 commented:
> Thanks again for this contribution, @nicedevil007 ! It looks really good, and thanks for your patience with my and @4d62 's constant edits, lol.
> 
> I know what you mean about the more you use git the more clear it becomes, so yeah this PR has been good practice I imagine. I am curious what you mentioned about modules, node modules? What was that exactly? Are you combining multiple repos using modules?
> 
> I think we can merge this now!

oh I just changed it to italic now everywhere whaaaaaaaaaa

nicedevil007 commented:
@4d62 @tanberry heya just wanted to ask you if I can do anything more to get this merged, want to raise a PR on actual as well to the right documentation page afterwards ;)

tanberry commented:
> earlier PR with ~ 56.000

Literally the stuff of nightmares!

tanberry commented:
Whoot AND yeehaw, it's merged! Thanks again, @nicedevil007.
